src/components/ConnectedEvent.js

```diff
@@ -5,17 +5,25 @@ import queryString from "query-string";
 import { setEvent } from "../actions/PageActions";
 
 class ConnectedEvent extends Component {
+  componentDidMount() {
+    this.setEventFromQueryParams()
+  }
+
   componentDidUpdate() {
+    this.setEventFromQueryParams()
+  }
+
+  render() {
+    return <Event event={this.props.event} />;
+  }
+
+  setEventFromQueryParams() {
```

Comment:
On mount and update because 
componentDidMount() doesn't work when clicking on the event link
componentDidUpdate() doesn't work when navigating to the url